examples/smcfilter.py

```diff
@@ -102,12 +99,50 @@ def main(args):
     logging.info("std: {}".format(z.variance ** 0.5))
 
 
+def tmc_run(args):
```

Comment:
Could you add some docstrings to `tmc_run()` and `smc_run` to inform the first-time reader that tmc is like smc but shares trajectories, and maybe adding some educational material describing similarities to other smc algorithms? This file seems like a good place for that material.

also nit: maybe call these `infer_smc`/`infer_tmc` or other descriptive name

Reply:
Since `TraceTMC_ELBO` isn't directly comparable to `SMCFilter` in functionality (e.g. no backward posterior sampling impemented, filtering vs smoothing) and is still pretty unstable, I'll just revert the changes to this example for now and make a more complete version once I've addressed those issues.